src/discussions/comments/data/selectors.js

```diff
@@ -1,10 +1,23 @@
 /* eslint-disable import/prefer-default-export */
-export const selectThreadComments = threadId => state => (state.comments.threadCommentMap[threadId] || []).map(
-  commentId => state.comments.comments[commentId],
+import { createSelector } from '@reduxjs/toolkit';
+
+const selectCommentsById = state => state.comments.commentsById;
+const mapIdToComment = (ids, comments) => ids.map(id => comments[id]);
+
+export const selectThreadComments = threadId => createSelector(
+  [
+    state => state.comments.commentsInThreads[threadId] || [],
+    selectCommentsById,
+  ],
+  mapIdToComment,
 );
 
-export const selectCommentResponses = commentId => state => (state.comments.commentResponsesMap[commentId] || []).map(
-  cId => state.comments.comments[cId],
+export const selectCommentResponses = commentId => createSelector(
+  [
+    state => state.comments.commentsInComments[commentId] || [],
```

Comment:
why do you need this empty array here I think the way normalization is handled in thunk this will never be undefined.

Reply:
If you refresh the page on a comment that hasn't loaded yet, this array will be empty. So instead of breaking the page, or needing to add logic for handling nulls everywhere, we return an empty array in that case.